docs/machine-learning/how-to-guides/explain-machine-learning-model-permutation-feature-importance-ml-net.md

```diff
@@ -0,0 +1,174 @@
+var sdcaEstimator = mlContext.Regression.Trainers.Sdca();
+
+// 6. Train machine learning model
+var sdcaModel = sdcaEstimator.Fit(preprocessedTrainData);
+```
+
+## Explain the model with Permutation Feature Importance (PFI)
+
+In ML.NET use the [`PermutationFeatureImportance`](xref:Microsoft.ML.PermutationFeatureImportanceExtensions) method for your respective task.
+
+```csharp
+ImmutableArray<RegressionMetricsStatistics> permutationFeatureImportance = 
+    mlContext
+        .Regression
+        .PermutationFeatureImportance(sdcaModel, preprocessedTrainData, permutationCount:3);
+```
+
+The result of using [`PermutationFeatureImportance`](xref:Microsoft.ML.PermutationFeatureImportanceExtensions) on the training dataset is an [`ImmutableArray`](xref:System.Collections.Immutable.ImmutableArray) of [`RegressionMetricsStatistics`](xref:Microsoft.ML.Data.RegressionMetricsStatistics) objects. [`RegressionMetricsStatistics`](xref:Microsoft.ML.Data.RegressionMetricsStatistics) provides summary statistics like mean and standard deviation for multiple observations of [`RegressionMetrics`](xref:Microsoft.ML.Data.RegressionMetrics) equal to the number of permutations specified by the `permutationCount` parameter.
+
+The importance or in this case the average decrease in R-squared metric calculated by [`PermutationFeatureImportance`](xref:Microsoft.ML.PermutationFeatureImportanceExtensions) can then be ordered from most important to least important.  
+
+```csharp
+// Order features by importance
+var featureImportanceMetrics =
+    permutationFeatureImportance
+        .Select((metric, index) => new { index, metric.RSquared })
+        .OrderByDescending(myFeatures => myFeatures.RSquared.Mean);
+
+Console.WriteLine("Feature\tPFI");
+
+foreach (var feature in featureImportanceMetrics)
+{
+    Console.WriteLine($"{featureColumnNames[feature.index],-20}|\t{feature.RSquared.Mean,10}");
+}
+```
+
+Printing the values for each of the features in `featureImportanceMetrics` would generate output similar to that below. Keep in mind that you should expect to see different results because these values vary based on the data that they are given.  
+
+| Feature | Change to R-Squared |
+|:--|:--:|
+NearWater           |   -0.000009
+HomeAge             |   -0.000125
+PercentPopulationLivingBelowPoverty|    -0.000148
+CommercialZones     |   -0.000518
+ResidentialZones    |   -0.001419
+ToxicWasteLevels    |   -0.003058
+CrimeRate           |   -0.003955
+TaxRate             |   -0.006024
+BusinessCenterDistance| -0.008286
+AverageRoomNumber   |   -0.010224
+StudentTeacherRatio |   -0.017148
+HighwayAccess       |   -0.038572
+
+Taking a look at the the five most important features for this dataset, the price of a house predicted by this model is influenced by its proximity to water, age of the home, percent of population living below poverty as well as the composition of commercial and residential zones.
```

Comment:
>ts proximity to water, age of the home, percent of population living below poverty as [](start = 135, length = 85)

i thought the ones with higher absolute value are the ones that affect it the most. Double-checking.

Reply:
[https://microsoft.sharepoint.com/teams/TLC/SitePages/Modes/Permutation%20Feature%20Importance.aspx](https://microsoft.sharepoint.com/teams/TLC/SitePages/Modes/Permutation%20Feature%20Importance.aspx)

The most important features are the ones at the bottom of the list. 

---
In reply to: [280125756](https://github.com/dotnet/docs/pull/11946#discussion_r280125756) [](ancestors = 280125756)

Reply:
Yes it should be the other way around, the greatest change indicates the greatest importance.

---
In reply to: [280127518](https://github.com/dotnet/docs/pull/11946#discussion_r280127518) [](ancestors = 280127518,280125756)

Reply:
@artidoro to confirm greatest absolute change. Is that correct?

Reply:
Yes, absolute change.

Reply:
Perfect. Thanks!

Reply:
The change should be negative :) since we are shuffling the feature values, we are basically guaranteed to get a lower score.

Reply:
#resolved

Reply:
#resolved